data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/DocumentBuilder.java

```diff
@@ -4,14 +4,14 @@
 
 public final class DocumentBuilder {
 
-    public static String build(final Event event, final String documentRootKey) {
+    public static String build(final Event event, final String documentRootKey, final String tagsKeyName) {
         if (documentRootKey != null && event.containsKey(documentRootKey)) {
             final String document = event.getAsJsonString(documentRootKey);
             if (document == null || !document.startsWith("{")) {
                 return String.format("{\"data\": %s}", document);
             }
             return document;
         }
-        return event.toJsonString();
+        return tagsKeyName == null ? event.toJsonString() : event.jsonBuilder().includeTags(tagsKeyName).toJsonString();
```

Comment:
I think the ideal usage is this:

```
return event.jsonBuilder()
  .includeTags(tagsKeyName)
  .toJsonString();
```

If the `includeTags(String)` method can take a `null` and provide the expected output, then this is solved for all sinks and they don't need to have this condition.